src/components/tag/Tag.tsx

```diff
@@ -89,15 +102,23 @@ const styles = StyleSheet.create({
  */
 export const Tag = ({ text, variant, testID }: Tag) => (
   <View testID={testID} style={styles.tag}>
-    <View style={styles.iconWrapper}>
-      <Icon
-        name={mapVariants[variant].iconName}
-        color={mapVariants[variant].iconColor}
-        size={IOTagIconSize}
-      />
-    </View>
-    <Text numberOfLines={1} ellipsizeMode="tail" style={styles.label}>
-      {text}
-    </Text>
+    {pipe(
+      mapVariants[variant],
+      O.fromNullable,
+      O.fold(
+        () => null,
+        ({ iconColor, iconName }) => (
+          <View style={styles.iconWrapper}>
+            <Icon name={iconName} color={iconColor} size={IOTagIconSize} />
+          </View>
+        )
+      )
+    )}
+    {mapVariants[variant] && text && <HSpacer size={8} />}
```

Comment:
Not a perfect implementation, but it's just an exception for this particular component. If other components start using this `6px` spacing value, we should consider adding it to the `V/HSpacer` spec.
```suggestion
    {mapVariants[variant] && text && <View style={styles.spacer} />}
```

Reply:
What about adding the value 6 to the spacer?

Reply:
According to the design specs, there is no `6` value in the spacing scale used for `H/VSpacer` components. Adopting it means adding that value to the `Layout` DS page, which gives the false impression that it's a widely accepted value. It's not, because it's only used here, for now.

Reply:
The `6` value is already added to the `IOSpacingScale` type, btw:
```ts
const IOTagIconMargin: IOSpacingScale = 6;
```